src/script/properties/PropertiesRepository.ts

```diff
@@ -138,8 +141,24 @@ export class PropertiesRepository {
     const isTelemetryPreferenceSet = this.getPreference(PROPERTIES_TYPE.PRIVACY.TELEMETRY_SHARING) !== undefined;
 
     if (!isTelemetryPreferenceSet) {
-      this.savePreference(PROPERTIES_TYPE.PRIVACY.TELEMETRY_SHARING, true);
-      this.publishProperties();
+      PrimaryModal.show(PrimaryModalType.CONFIRM, {
+        text: {title: t('dataSharingModalTitle'), message: t('dataSharingModalDescription')},
+        primaryAction: {
+          text: t('dataSharingModalAgree'),
+          action: () => {
+            this.savePreference(PROPERTIES_TYPE.PRIVACY.TELEMETRY_SHARING, true);
+            this.publishProperties();
```

Comment:
Just a minor suggestion, would it be better if we create a function like:
```
const toggleTelemetrySharing = (value) => {
  this.savePreference(PROPERTIES_TYPE.PRIVACY.TELEMETRY_SHARING, value);
  this.publishProperties();
}
```
and use in both primary and secondary action?